kybra/compiler/kybra_post_install/src/main.rs

```diff
@@ -0,0 +1,402 @@
+            &install_output.stderr,
+        )));
+    }
+
+    Ok(())
+}
+
+fn remove_permissions(
+    canister_name: &str,
+    dfx_network: &str,
+    canister_id: &str,
+    canister_already_its_own_controller: bool,
+) -> Result<(), String> {
+    if canister_already_its_own_controller {
+        return Ok(());
+    }
+
+    let remove_controller_output = Command::new("dfx")
+        .arg("canister")
+        .arg("update-settings")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg("--remove-controller")
+        .arg(canister_id)
+        .arg(canister_name)
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx command"))?;
+
+    if !remove_controller_output.status.success() {
+        return Err(create_error_string(&String::from_utf8_lossy(
+            &remove_controller_output.stderr,
+        )));
+    }
+
+    Ok(())
+}
+
+fn upload_chunk(
+    name: &str,
+    canister_name: &str,
+    bytecode_chunk: &Vec<u8>,
+    canister_method_name: &str,
+    dfx_network: &str,
+    chunk_number: usize,
+    chunk_total: usize,
+) -> Result<(), String> {
+    let blob_string = vec_u8_to_blob_string(bytecode_chunk);
+
+    let mut temp_file =
+        NamedTempFile::new().map_err(|_| create_error_string("Failed to create temporary file"))?;
+    temp_file
+        .write_all(blob_string.as_bytes())
+        .map_err(|_| create_error_string("Failed to write data to temporary file"))?;
+
+    let output = Command::new("dfx")
+        .arg("canister")
+        .arg("call")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg(canister_name)
+        .arg(canister_method_name)
+        .arg("--argument-file")
+        .arg(temp_file.path())
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx command"))?;
+
+    let chunk_number = chunk_number + 1;
+
+    if output.status.success() {
+        println!(
+            "{}",
+            format!("Uploading {name} chunk {chunk_number}/{chunk_total}")
+        );
+    } else {
+        return Err(create_error_string(&String::from_utf8_lossy(
+            &output.stderr,
+        )));
+    }
+
+    Ok(())
+}
+
+fn split_into_chunks(data: Vec<u8>, chunk_size: usize) -> Vec<Vec<u8>> {
+    let mut chunks = Vec::new();
+    let mut start = 0;
+    let data_len = data.len();
+
+    while start < data_len {
+        let end = usize::min(start + chunk_size, data_len);
+        chunks.push(data[start..end].to_vec());
+        start = end;
+    }
+
+    chunks
+}
+
+fn vec_u8_to_blob_string(data: &[u8]) -> String {
+    let mut result = String::from("(blob \"");
+    for &byte in data {
+        result.push_str(&format!("\\{:02X}", byte));
+    }
+    result.push_str("\")");
+    result
+}
+
+fn error_to_string(e: &dyn std::error::Error) -> String {
+    format!("{ERROR_PREFIX}: {}", e.to_string())
+}
+
+fn create_error_string(message: &str) -> String {
+    format!("{ERROR_PREFIX}: {message}")
+}
```

Comment:
This file seems to have grown quite large. I would recommend trying to break it down into smaller files. These functions at the bottom clearly seems like a utils section. So yeah, just try to break this up.